lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2818,3 +2819,17 @@ void CoverOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                           MLIRContext *context) {
   results.add(canonicalizeImmediateVerifOp<CoverOp, /* EraseIfZero = */ true>);
 }
+
+//===----------------------------------------------------------------------===//
+// InvalidValueOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult InvalidValueOp::canonicalize(InvalidValueOp op,
+                                           PatternRewriter &rewriter) {
+  // Remove `InvalidValueOp`s with no uses.
```

Comment:
This can be a pattern too.  There are special Pat keywords for erase-and-do-nothing-else.

Reply:
Can't find the machinery for that. The `PatternBase.td` seems pretty basic at the moment. All the utilities seem to focus on replacing results with all kinds of fancy expressions.